blog/2025-06-25-introducing-flet-1-0-alpha.md

```diff
@@ -0,0 +1,502 @@
+
+* Blocking calls like `time.sleep()` will freeze the UI. Instead of `time.sleep()` use `async def` event handlers and using `await asyncio.sleep()` for delays.
+* Switch to async APIs whenever possible.
+* For CPU-bound tasks, offload them to threads using `asyncio.to_thread(...)`.
+
+🚧 Example with CPU-bound method updating progress bar 🚧
+
+### Async control methods
+
+All controls' get- and set- methods are async now.
+
+Methods that do not return any results have fire-and-forget sync wrappers.
+
+🚧 Documentation is in progress 🚧
+
+### `ft.run()` replaces `ft.app()`
+
+Method arguments stay the same. A new `before_main` argument added (see above).
+
+### `Page` split
+
+`Page` split into `Page` and `PageView`.
+
+To support Flet embedding with multi-views.
+
+It's not a breaking-change per-se if you just continue to use `page` instance methods or properties.
+
+🚧 Documentation is in progress 🚧
+
+### Dialogs
+
+To display a dialog, banner, snack bar, drawer, or any similar popup control, use `page.show_dialog(dialog_control)` instead of `page.open()`.
+
+To close the topmost popup, use `page.pop_dialog()`.
+
+### Drawers
+
+`page.drawer` and `page.end_drawer` were removed.
+
+Use `NavigationDrawer.position` property and then `page.show_dialog()` to display drawer instead of `page.drawer` and `page.end_drawer`.
+
+### FilePicker
+
+FilePicker is now a service and must be added to `page.services` list to work.
+
+API re-worked to provide async methods immediatly returning dialog results without using "result" event handlers.
+
+```py
+files: list[FilePickerFile] = await file_picker.pick_files_async(allow_multiple=True)
+file_name: str = await file_picker.save_file_async()
+dir_name: str = await file_picker.get_directory_path_async()
+```
+
+Full examples:
+
+* [All dialogs](https://github.com/flet-dev/examples/blob/v1/python/controls/utility/file-picker/file-picker-all-modes.py)
+* [Upload](https://github.com/flet-dev/examples/blob/v1/python/controls/utility/file-picker/file-picker-upload-progress.py)
+
+### HapticFeedback
+
+`HapticFeedback` is now a Service and must be added to `page.services` to work.
+
+# SemanticsService
+
+`SemanticsService` is now a Service and must be added to `page.services` to work.
+
+# ShakeDetector
+
+`ShakeDetector` is now a Service and must be added to `page.services` to work.
+
+### Clipboard
+
+Instead of `page.set_clipboard()` use `page.clipboard.set_async()`.
+
+Instead of `page.get_clipboard()` use `page.clipboard.get_async()`.
+
+### Client storage
+
+"Client storage" is now "Shared Preferences".
+
+`page.client_storage` property renamed to `page.shared_preferences`.
+
+### Scrollables
+
+In scrollable controls `on_scroll_interval` property renamed to `scroll_interval`.
+
+### Buttons
+
+All buttons: no `text` property, use `content` instead.
+
+### Control ID is integer
+
+`e.target` is a number now, not a string.
+
+## The new Architeccture
```

Comment:
Let's move this section a little more to the top as it's more important. Maybe somewhere before "breaking changes".